mailpoet/lib/Automation/Integrations/MailPoet/Subjects/SubscriberSubject.php

```diff
@@ -132,10 +154,85 @@ function (SubscriberPayload $payload) {
           ],
         ]
       ),
+      new Field(
+        'mailpoet:subscriber:subscription-source',
+        Field::TYPE_ENUM,
+        __('Subscription source', 'mailpoet'),
+        function (SubscriberPayload $payload) {
+          return $payload->getSubscriber()->getSource();
+        },
+        [
+          'options' => [
+            [
+              'id' => 'api',
+              'name' => __('API', 'mailpoet'),
+            ],
+            [
+              'id' => 'form',
+              'name' => __('Form', 'mailpoet'),
+            ],
+            [
+              'id' => 'unknown',
+              'name' => __('Unknown', 'mailpoet'),
+            ],
+            [
+              'id' => 'imported',
+              'name' => __('Imported', 'mailpoet'),
+            ],
+            [
+              'id' => 'administrator',
+              'name' => __('Administrator', 'mailpoet'),
+            ],
+            [
+              'id' => 'wordpress_user',
+              'name' => __('WordPress user', 'mailpoet'),
+            ],
+            [
+              'id' => 'woocommerce_user',
+              'name' => __('WooCommerce user', 'mailpoet'),
+            ],
+            [
+              'id' => 'woocommerce_checkout',
+              'name' => __('WooCommerce checkout', 'mailpoet'),
+            ],
+          ],
+        ]
+      ),
+      new Field(
+        'mailpoet:subscriber:last-subscribed-at',
+        Field::TYPE_DATETIME,
+        __('Subscribed date', 'mailpoet'),
+        function (SubscriberPayload $payload) {
+          return $payload->getSubscriber()->getLastSubscribedAt();
+        }
+      ),
+      new Field(
+        'mailpoet:subscriber:tags',
+        Field::TYPE_ENUM_ARRAY,
+        __('Tags', 'mailpoet'),
+        function (SubscriberPayload $payload) {
+          $value = [];
+          foreach ($payload->getSubscriber()->getSubscriberTags() as $subscriberTag) {
+            $tag = $subscriberTag->getTag();
+            if ($tag) {
+              $value[] = $tag->getId();
+            }
+          }
+          return $value;
+        },
+        [
+          'options' => array_map(function ($tag) {
+            return [
+              'id' => $tag->getId(),
+              'name' => $tag->getName(),
+            ];
+          }, $this->tagRepository->findAll()),
```

Comment:
I dont think we have complete `CRUD` yet for tags, but maybe it would make sense to exclude deleted tags?

Reply:
Tags don't seem to have a soft delete functionality yet: https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/Entities/TagEntity.php

Reply:
Not super important, but I am sure, we will forget about this, once we get such functionality. Something like a `//ToDo` comment maybe, so that we stumble upon it at some point and think "Ah, wait!"